src/expr.cpp

```diff
@@ -4141,6 +4141,13 @@ llvm::Value *FunctionCallExpr::GetValue(FunctionEmitContext *ctx) const {
         if (isInvoke) {
             return ctx->InvokeSyclInst(callee, ft, argVals);
         } else {
+            // if callee is llvm.masked.gather then cast with inttoptr the first arg to PtrVectorType
+            if (callee->getName().starts_with("llvm.masked.gather")) {
+                argVals[0] = ctx->IntToPtrInst(argVals[0], LLVMTypes::PtrVectorType);
+            }
+            if (callee->getName().starts_with("llvm.masked.scatter")) {
+                argVals[1] = ctx->IntToPtrInst(argVals[1], LLVMTypes::PtrVectorType);
+            }
```

Comment:
There is nothing wrong with the code except that it looks like a hack in this place :)
It should be done earlier when we check the function arguments (line 4111 or earlier?). Also can we do this through `TypeConvertExpr`?

Reply:
> There is nothing wrong with the code except that it looks like a hack in this place :) It should be done earlier when we check the function arguments (line 4111 or earlier?). Also can we do this through `TypeConvertExpr`?

4111 or earlier and `TypeConvertExpr` operates in ISPC types whereas we adjust only LLVM types here.

Reply:
I see.